src/plugins/intel_cpu/src/infer_request.cpp

```diff
@@ -373,6 +373,46 @@ void LegacyInferRequest::SetBatch(int new_batch) {
     }
 }
 
+void LegacyInferRequest::changeDefaultPtr() {
+    // renew external pointers before infer
+    const auto &inMap = graph->inputNodesMap;
+    for (auto &it : inMap) {
+        const auto &name = it.first;
+        auto itr = externalPtr.find(name);
+        if (itr != externalPtr.end() && itr->second != _inputs[name]->buffer()) {
+            itr->second = _inputs[name]->buffer();
+        }
+    }
+    const auto &outMap = graph->outputNodesMap;
+    for (auto &it : outMap) {
+        const auto &name = it.first;
+        auto itr = externalPtr.find(name);
+        if (itr != externalPtr.end() && itr->second != _outputs[name]->buffer()) {
+            itr->second = _outputs[name]->buffer();
+        }
+    }
+    InferRequestBase::changeDefaultPtr();
+}
+void LegacyInferRequest::tuneInputDesc(const std::string &name, InferenceEngine::TensorDesc &desc) {
```

Comment:
Since `tuneInputDesc` and `tuneOutputDesc` are now used only once, it does make sense to remove the methods and inline the code in place where they are called (simply revert extracting those code frames to the separate methods).